### DISCUSSION TITLE
Proposal: Extend this repo's scope to include OPML

### DISCUSSION THREAD

**Discussion by saerdnaer** [Podcastindex-org/podcast-namespace#478] [Spec Proposal]
I think we should also provide good example representations of podcast collections / item recommendation lists in OPML, not only in RSS or custom JSON.

#  What is OPML 

[OPML](https://en.wikipedia.org/wiki/OPML) (Outline Processor Markup Language) comes from the same person as RSS 0.92 / 2.0 - Dave Winer.
As its long name suggest, OPML a XML dialect to describe outlines – but became more popular as as simple way to list RSS feeds, e.g. a blog roll or to transfer your subscriptions from one podcast app to another. 
This is a separate section 'subscription lists' in the OPML spec: 
* http://dev.opml.org/spec2.html#subscriptionLists

As already [mentioned in another discussion](https://github.com/Podcastindex-org/podcast-namespace/issues/240#issuecomment-836693793) there is also a JSON representation:
* http://scripting.com/2016/10/11/convertOpmlToJson.html
* http://scripting.com/code/opmltojson/?url=http://www.bbc.co.uk/podcasts.opml

I can also recommend to export your subscriptions via OPML, convert it to CSV via http://opml2csv.com and sort them in your favourite spreadsheet tool, as I did in Sep 2018, see below [1].  :-)

Broadcasters / Podcast networks already provide their list of podcasts als OPML e.g. http://podcasts.files.bbci.co.uk/podcasts.opml [[via BBC radiolabs Blog](http://www.bbc.co.uk/blogs/radiolabs/2008/04/opml_feed_of_podcasts.shtml)]

# Reasoning 

As of today nearly every podcast app supports import and export of subscriptions, but all other information like how many episodes you listened are lost. We should extend OPML and document common practises like we do with RSS.

This is not thought as apposing proposal to the linked issues, but should be seen as an extension to the existing proposals to make it easier for users and app developers to implement ans use these ideas.

TBC…

# Misc

Related: https://github.com/Podcastindex-org/podcast-namespace/issues/240 https://github.com/Podcastindex-org/podcast-namespace/issues/363

[1]
<img width="1640" alt="image" src="https://user-images.githubusercontent.com/40266/165087872-1ff24236-403f-46e7-8797-c595e3c07e9d.png">

**Comment by daveajones** [comment]
I would also like to propose the namespace to be used in OPML for the basis of the channels tag work.

**Comment by matthewruzzi** [comment]
It would be very helpful to be able to export information about individual episodes in OPML. Currently switching between podcast apps requires me to manually go through every episode I have listened to and mark it as played.

**Comment by samsethi** [comment]
OPML is 1.0 legacy and should be dead by now.  Maybe only retained for backward compatibility.  I thought we had moved on to Remote Item based Podrolls and Publisher feeds.  

TrueFans "imports" these feeds from an RSS feed and can also do so from standalone feeds, as they are fully formed RSS feeds.  We also offer creators/fans the option 'export' these feeds. - see image below 

This is the [Wondery Publisher Feed ](https://truefans.fm/publishers/wondery)with an export option using Remote Items. 

<img width="1481" alt="Fullscreen_14_12_2024__19_38" src="https://github.com/user-attachments/assets/997ab504-1a7c-4771-8f14-d6f628a75b9f" />

**Reply by matthewruzzi** [reply]
@samsethi What would need to be done to allow transferring information about individual episodes between podcast apps?

**Reply by samsethi** [reply]
Hi Matthew what information do you want to transfer?  If it's simply my library of podcasts I am subscribed to then OPML is one method and Podrolls, Playlists and/or Publisher Feeds are another open exchange method using Remote Items and GUIDs. 

If it is more detailed information about the user's activity with an episode such as Listen Time, Percent Completed and Value Paid then we show this activity data to all users on their profile page. We can also show the total activity data for a podcast but we give that to the creators in their admin dashboard. 

e.g https://truefans.fm/fans/sam?tab=activities 
<img width="1485" alt="TrueFans___Sam_Sethi" src="https://github.com/user-attachments/assets/839f76ba-285a-413f-b28b-141037df5ed5" />

**Reply by matthewruzzi** [reply]
The primary thing that I would want to be able to transfer between apps is which episodes are played or not. Overcast has a way to export OPML with this information but "no apps are currently known to import this non-standard format". 

![Screenshot 2025-02-25 at 3 47 47 PM](https://github.com/user-attachments/assets/85c2073e-0d9f-4d2a-b70f-bbfeab8620fd)

**Reply by keunes** [reply]
@matthewruzzi With [Open Podcast API](https://openpodcastapi.org/) we're actually working on a standard to exchange podcast usage data, also for podcasts but mainly related to episodes: has this episode been listened to, favourited, otherwise bookmarked, etc. Getting to a workable data portability exchange is hard in an environment where apps already have their own data models and approaches. So we'd very much welcome other contributions here!

**Reply by daveajones** [reply]
@keunes Is there an OPML component to your Open Podcast API work or is your group purely focused on the online API part of things?
